src/server.py

```diff
@@ -120,6 +123,7 @@ def broadcast(self, summary, encoded_message):
                 player_socket.sendall(encoded_message)
                 logger.info(
                     "Broadcasted {%s} to Player {%s}", summary, name)
+                _ack = s.decode_ack(player_socket.recv(1024))  # Receiving ack
```

Comment:
nitpicking - this is essentially serializing the communication (no concurrency) - i'm gonna look into this after merge.